### ISSUE TITLE
MarkerView anchor after resizing View

### ISSUE THREAD

**Issue by Larnoo** [mapbox/mapbox-gl-native#8798]
**Platform:** Android
**Mapbox SDK version:** 5.0.0

### Steps to trigger behavior

 1. the marker view anchor is 0.5f, 1.0f. 
 2. When click marker view, at MarkerViewAdapter->onSelect, I update the convertView size.
 3. the marker view anchor is not at bottom-center!

deselect
![screenshot_2017-04-24-12-12-13](https://cloud.githubusercontent.com/assets/15139682/25322296/34243a32-28e8-11e7-9ed4-318fcf8b7776.png)
selected
![screenshot_2017-04-24-12-13-38](https://cloud.githubusercontent.com/assets/15139682/25322309/54d973aa-28e8-11e7-9f9c-1e1928165212.png)

**Comment by tobrun** [comment]
Would you be able to retest with using 5.0.2? this version contains a fix for anchoring in https://github.com/mapbox/mapbox-gl-native/pull/8519. Thank you for reaching out and reporting your issue.

**Comment by tobrun** [comment]
Going to close this as this was tested with an older version without the fix from #8519. Feel free to let us know if this issue still an issue with 5.0.2. Thank you for reaching out. Closing for now.

**Comment by Larnoo** [comment]
@tobrun 
I use Mapbox SDK v5.0.2, but I have the same problem.
The patch can't solve my problem. 

Becasue my MarkerView is not a `ImageView` but a `TextView`.

When MarkerView selected, I update `TextView` with larger text size, but the MakerView `width` and `height` is not update. 

So the MarkerView anchor is incorrect!

```
  /**
   * Update and invalidate the MarkerView icon.
   *
   * @param markerView the marker view to updates.
   */
  public void updateIcon(@NonNull MarkerView markerView) {
    View convertView = markerViewMap.get(markerView);
    if (convertView != null && convertView instanceof ImageView) {
      ((ImageView) convertView).setImageBitmap(markerView.getIcon().getBitmap());
      markerView.invalidate();
    }
  }

```

**Comment by tobrun** [comment]
Thank you for retesting. Was able to reproduce the issue. Linking https://github.com/mapbox/mapbox-gl-native/issues/6805 as the solution for that issue will require introducing wrapper ViewGroup. This wrapper could be used to apply the anchoring on apply a gravity to the View of the MarkerView. Reopening.

**Comment by Guardiola31337** [comment]
Hey 👋  @Larnoo 
First of all, thanks a lot for reaching out and report your problem.

This should be fixed with #9282. Would you be able to retest using `5.1.0-beta.5` or the `SNAPSHOT` to verify if it's fixed? Thanks again!

**Comment by Guardiola31337** [comment]
Going ahead and closing here.

@Larnoo thanks for using Mapbox and feel free to open new tickets if you find any other issues 😉
